GTMAppAuthSwift/Sources/GTMKeychainManager.swift

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+/// A class managing the reading and saving of `GTMAppAuthFetcherAuthorization` to the Keychain.
+@objc public final class GTMKeychainManager: NSObject {
```

Comment:
Why introduce a new class vs extend GTMAppAuthFetcherAuthorization as the ObjC keychain category does?

Reply:
I want to be able to test `GTMAppAuthFetcherAuthorization`s keychain reads, writes, and removes and do not want `GTMKeychain`, `KeychainHelper`, or `KeychainWrapper` to be `public`. If `GTMAppAuthFetcherAuthorization` interacts directly with the keychain to read, save, and remove authorizations, then we will need to pass it a fake (e.g., via a class method) that (at least) conforms to `KeychainHelper` to verify these interactions. That would require making (at least one of) the above types public. I'd prefer to not do that because it feels like leaking implementation details into the public API solely for the purposes of testing.

Happy to hear other ideas on how this can organized and tested without marking things as `public` unnecessarily.

Reply:
If we extend GTMAppAuthFetcherAuthorization we can consider adding a private variable keychainHelper: KeychainHelper.
In the public init keychainHelper = KeychainWrapper().
We add another private init for testing in which keychainHelper = KeychainHelperFake().
In this way we can extend GTMAppAuthFetcherAuthorization without exposing keychain classes.

Reply:
@Alex-4-Git and I followed up on his comment offline. I think we can disregard.

Reply:
@mdmathias I believe all keychain operations in the existing `GTMAppAuthFetcherAuthorization+Keychain` go through `GTMKeychain`, so I would assume that's what we would want to fake for testing.  To address the problem of how to provide a fake `GTMKeychain` given that we can cleanly add storage via an extension, have you considered simply merging `GTMAppAuthFetcherAuthorization+Keychain` into `GTMAppAuthFetcherAuthorization`?  We discussed this route briefly a few weeks ago.  I'm not sure that the category is currently providing much value, if we can simplify by merging, great, and no API surface impact, which aligns with the goals of our translation.  You'd then have more options for supplying a fake `GTMKeychain`.

@Alex-4-Git extensions in Swift seem _at least_ as capable as categories in ObjC.  I don't think there's a Swift technical limitation that's preventing us from translating to an extension here, testing considerations aside.

Reply:
Resolving this as it was addressed in another pr.